.github/workflows/eval.yml

```diff
@@ -85,11 +85,13 @@ jobs:
         uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72 # v30
 
       - name: Evaluate the ${{ matrix.system }} output paths for all derivation attributes
+        env:
+          MATRIX_SYSTEM: ${{ matrix.system }}
         run: |
           nix-build nixpkgs/ci -A eval.singleSystem \
-            --argstr evalSystem ${{ matrix.system }} \
+            --argstr evalSystem "$MATRIX_SYSTEM" \
             --arg attrpathFile ./paths/paths.json \
-            --arg chunkSize 10000
+            --arg chunkSize 9000
```

Comment:
I am not sure the infinite recursion we are seeing has something to do with memory depletion. There was still 1GB memory free and also plenty of swap. We should upgrade to nix 2.24 because there detection for infinite recursion is much better as it doesn't use the segfault handler.

Reply:
Can we first just merge the first commit: https://github.com/NixOS/nixpkgs/pull/357800